src/webhooks/handleSetupIntentSucceded.ts

```diff
@@ -0,0 +1,18 @@
+import Stripe from 'stripe';
+import { PaymentService } from '../services/PaymentService';
+import { UserType } from '../core/users/User';
+
+export default async function handleSetupIntentSucceded(
+  setupIntent: Stripe.SetupIntent,
+  paymentService: PaymentService,
+): Promise<void> {
+  const customerId = setupIntent.customer as string;
+  const paymentMethodId = setupIntent.payment_method as string;
+  const setupIntentMetadata = setupIntent.metadata as Stripe.Metadata;
+  
+  const userType = setupIntentMetadata?.userType as UserType;
+
+  if ([UserType.Individual, UserType.Business].includes(userType)) {
+    await paymentService.updateSubscriptionPaymentMethod(customerId, paymentMethodId, userType);
```

Comment:
Are we sure that this [event](https://docs.stripe.com/api/setup_intents) is reliable enough to assume the payment is collected, thus, updating the payment method?

Reply:
I think this event is enough for that @sg-gs. The event is triggered when the setup intent has been completed: https://docs.stripe.com/api/events/types#event_types-setup_intent.succeeded. 

From what I understand now, when creating a setup intent you get the payment method id which is what we use to update the payment method.

Reply:
@sg-gs 
Thanks for answering @masterprog-cmd . Yes, correct, currently when we generate a Stripe Setup Intent (`/setup-intent`) we will be adding the `userType` in the metadata to know which subscription to work with when receiving a setupIntent event.

If there is no matching `userType` in the metadata, it means that the call for this event was made in another way and not from our endpoint side through the front end.

Once identified, we will then update the subscription of the corresponding user.